Codes/SlippiReplays/Record.cpp

```diff
@@ -23,164 +23,90 @@ namespace ReplaysLogic {
-                    positionsPacket.Send();
-
-                    EXIPacket fighterPacket = EXIPacket(EXIStatus::STARTFIGHTER, fighterInfo, 1 * sizeof(int));
-                    fighterPacket.Send();
-                    delete[] positions;
-                    delete[] fighterInfo;
+                    entryFrame = false;
+
+                    auto* startReplay = new StartReplay();
+
+                    const GameGlobal *game = GAME_GLOBAL;
+
+                    mtRand *rand = DEFAULT_MT_RAND;
+                    mtRand *otherRand = OTHER_MT_RAND;
+                    startReplay->randomSeed = rand->seed;
+                    startReplay->otherRandomSeed = otherRand->seed;
+
+                    startReplay->stage = game->globalModeMelee->stageID;
+                    startReplay->numPlayers = FIGHTER_MANAGER->getEntryCount();
+                    for(int i = 0; i < FIGHTER_MANAGER->getEntryCount(); i++)
+                    {
+                        auto fighter = FIGHTER_MANAGER->getFighter(FIGHTER_MANAGER->getEntryIdFromIndex(i));
+                        startReplay->players[i].fighterKind = fighter->getFtKind();
+                        auto xPos = fighter->modules->postureModule->xPos;
+                        auto yPos = fighter->modules->postureModule->yPos * -1;
+                        auto zPos = fighter->modules->postureModule->zPos;
+
+                        startReplay->players[i].startPlayer.xPos = xPos;
+                        startReplay->players[i].startPlayer.yPos = yPos;
+                        startReplay->players[i].startPlayer.zPos = zPos;
+                    }
+
+                    EXIPacket startReplayPacket = EXIPacket(EXIStatus::START_REPLAYS_STRUCT, startReplay, sizeof(StartReplay));
+                    startReplayPacket.Send();
+                    delete startReplay;
                 }
-            }
-            ftManager *fighter = FIGHTER_MANAGER;
-            const GameGlobal *game = GAME_GLOBAL;
-            itemIdName *itemIds;
-            u16 *itemVariants;
-            u32 *gameInfo = new u32[2];
-            int sizeOfItems = ITEM_MANAGER->baseItemArrayList.size();
-
-            gameInfo[0] = game->gameFrame->frameCounter;
-            gameInfo[1] = game->gameFrame->persistentFrameCounter;
-
-            EXIPacket gamePacket = EXIPacket(EXIStatus::GAME, gameInfo, 2 * sizeof(u32));
-            gamePacket.Send();
-            delete[] gameInfo;
-
-            if(sizeOfItems > 0) {
-                itemIds = new itemIdName[sizeOfItems];
-                itemVariants = new u16[sizeOfItems];
-                for (int i = 0; i < sizeOfItems; i++) {
-                    auto item = ITEM_MANAGER->baseItemArrayList.at(i);
-                    itemIds[i] = (*item)->getItKind();
-                    itemVariants[i] = (*item)->getItVariation();
+                else
+                {
+                    ftManager *fighter = FIGHTER_MANAGER;
```

Comment:
@JaredWhiteOne I would rename this to `fighterManager` instead. it misleads to what the object really is.